include/picongpu/plugins/common/openPMDDefaultExtension.hpp

```diff
@@ -46,19 +46,43 @@ namespace picongpu
             auto getADIOSExtension = []()
             {
                 auto availableExtensions = ::openPMD::getFileExtensions();
-                for(auto const& ext : availableExtensions)
-                {
-                    if(ext == "bp4")
-                    {
-                        return "bp4";
-                    }
-                }
                 /*
-                 * BP4 engine is always available in all supported ADIOS2 versions,
-                 * but the bp4 extensions is new in openPMD, so we might need to
-                 * fallback to "bp".
+                 * Using macros for detecting the ADIOS2 version is not very nice since the openPMD-api can be
+                 * recompiled against different backends and backend versions without affecting the ABI. Upgrading
```

Comment:
```suggestion
                 * recompiled against different backends and backend versions without affecting the API. Upgrading
```

Reply:
This is no typo, I'm referring to the application binary interface of openPMD-api. The API ideally does not change across different versions, but recompiling openPMD with different backends ideally will not even change the ABI, meaning that you only have to re-link, but not recompile PIConGPU.
Under these circumstances it's not really ideal to rely on compile-time information about the backends like it's done here. But it's only about the default values, so it's mostly fine still.

Reply:
Thanks for the clarification :+1: 
Just for others who are not aware of this abbreviation (like me):
*ABI = Application Binary Interface*